Packs/BastilleNetworks/Integrations/BastilleNetworks/BastilleNetworks.py

```diff
@@ -0,0 +1,364 @@
+                                    event_id=event_id)
+
+    readable_fields = ['event_id', 'area', 'zone_name', 'tags', 'device_info',
+                       'emitter', 'first_seen', 'last_seen']
+
+    raw_response = copy.deepcopy(events)
+    context_events, readable_events = parse_events(events, readable_fields)
+
+    readable_output = tableToMarkdown('Zone Events', readable_events, readable_fields)
+    context_output = {'Bastille.ZoneEvent(val.event_id == obj.event_id)': context_events}
+
+    return readable_output, context_output, raw_response
+
+
+def get_device_events_command(client, args):
+    """ Command to fetch device detection events """
+
+    site, concentrator, map = get_site_params()
+    protocol, since, until, limit, tags, event_id = get_event_args(args)
+    transmitter_id = args.get('transmitter_id')
+
+    events = client.get_device_events(site=site, concentrator=concentrator, map=map,
+                                      protocol=protocol, since=since, until=until,
+                                      limit=limit, tags=tags, event_id=event_id,
+                                      transmitter_id=transmitter_id)
+
+    readable_fields = ['event_id', 'area', 'tags', 'device_info', 'emitter',
+                       'first_seen', 'last_seen']
+
+    raw_response = copy.deepcopy(events)
+    context_events, readable_events = parse_events(events, readable_fields)
+
+    readable_output = tableToMarkdown('Device Events', readable_events, readable_fields)
+    context_output = {
+        'Bastille.DeviceEvent(val.event_id == obj.event_id)': context_events}
+
+    return readable_output, context_output, raw_response
+
+
+def add_device_tag_command(client, args):
+    """ Command to add tag to an existing admin devices entry """
+
+    site, concentrator, map = get_site_params()
+
+    transmitter_id = args.get('transmitter_id')
+    tag = args.get('tag')
+
+    result = client.add_device_tag(site=site, concentrator=concentrator, map=map,
+                                   transmitter_id=transmitter_id, tag=tag)
+
+    if 'status' not in result:
+        return_error('Failed to add device tag')
+
+    return result['status'], {}, result
+
+
+def remove_device_tag_command(client, args):
+    """ Command to remove tag from an existing admin devices entry """
+
+    site, concentrator, map = get_site_params()
+
+    transmitter_id = args.get('transmitter_id')
+    tag = args.get('tag')
+
+    result = client.remove_device_tag(site=site, concentrator=concentrator, map=map,
+                                      transmitter_id=transmitter_id, tag=tag)
+
+    if 'status' not in result:
+        return_error('Failed to remove device tag')
+
+    return result['status'], {}, result
+
+
+def fetch_incidents(client, last_run):
+    """ Callback to fetch incidents periodically """
+
+    last_fetch_time = last_run.get('last_fetch', datetime.utcnow().timestamp() - 60)
```

Comment:
```suggestion
    last_fetch_time = last_run.get('last_fetch', datetime.utcnow().timestamp() - NEW_PARAM)
```

Currently, if a user wants to reset the fetch incidents he will only be able to get the events from a single minute back. 
We should add a new integration param stating how long back (in seconds) should the first fetch go - this will enable the user finer control in case an error occurs of he wants to reset the fetch for any reason.